### ISSUE TITLE
[BUG] Fire-and-forget method invocation does not work, if method argument is not a trivial type 

### ISSUE THREAD

**Issue by anowak-ct** [jobrunr/jobrunr#158]
**Are you running the latest version of JobRunr?**
Yes 

**Describe the bug**
This will throw an exception:

```
@Inject
private JobScheduler jobScheduler;
MyPoJo pojo = new MyPoJo(1,"test"));
jobScheduler.enqueue(() -> myService.doWork(pojo));
```
`MyPoJo` is just a simple POJO class with an int and String property, incl all args and no args constructor and getter and setters.

Exception:
```
 java.lang.NullPointerException: null
     at org.jobrunr.utils.reflection.autobox.Autoboxer.autobox(Autoboxer.java:29) ~[jobrunr-3.1.2.jar!/:na]
     at org.jobrunr.utils.reflection.ReflectionUtils.autobox(ReflectionUtils.java:288) ~[jobrunr-3.1.2.jar!/:na]
     at org.jobrunr.jobs.details.instructions.VisitMethodInstruction.getParametersUsingParamTypes(VisitMethodInstruction.java:40) ~[jobrunr-3.1.2.jar!/:na]
     at org.jobrunr.jobs.details.instructions.InvokeSpecialInstruction.invokeInstruction(InvokeSpecialInstruction.java:27) ~[jobrunr-3.1.2.jar!/:na]
     at org.jobrunr.jobs.details.instructions.AbstractJVMInstruction.invokeInstructionAndPushOnStack(AbstractJVMInstruction.java:18) ~[jobrunr-3.1.2.jar!/:na]
     at org.jobrunr.jobs.details.JobDetailsFinderContext.invokeInstructions(JobDetailsFinderContext.java:92) ~[jobrunr-3.1.2.jar!/:na]
     at org.jobrunr.jobs.details.JobDetailsFinderContext.getJobDetails(JobDetailsFinderContext.java:70) ~[jobrunr-3.1.2.jar!/:na]
     at org.jobrunr.jobs.details.JobDetailsAsmGenerator$JobDetailsFinder.getJobDetails(JobDetailsAsmGenerator.java:229) ~[jobrunr-3.1.2.jar!/:na]
     at org.jobrunr.jobs.details.JobDetailsAsmGenerator.toJobDetails(JobDetailsAsmGenerator.java:31) ~[jobrunr-3.1.2.jar!/:na]
     at org.jobrunr.scheduling.JobScheduler.enqueue(JobScheduler.java:103) ~[jobrunr-3.1.2.jar!/:na]
     at org.jobrunr.scheduling.JobScheduler.enqueue(JobScheduler.java:85) ~[jobrunr-3.1.2.jar!/:na]
     at com.test.MyService.doWork(MyService.java:215) ~[classes!/:0.0.1-SNAPSHOT]
<snip>     
     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
     at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
     at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[app.jar:0.0.1-SNAPSHOT]
     at org.springframework.boot.loader.Launcher.launch(Launcher.java:108) ~[app.jar:0.0.1-SNAPSHOT]
     at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[app.jar:0.0.1-SNAPSHOT]
     at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88) ~[app.jar:0.0.1-SNAPSHOT]
```

**Environment**
I'm using JobRunr version: 3.1.2
I'm running on JRE / JDK: OpenJDK 64-Bit Server VM Corretto-11.0.11.9.1 (build 11.0.11+9-LTS, mixed mode)
I'm using the following StorageProvider (e.g. Oracle / MongoDB / ...): none

**How to Reproduce**
see above

**Expected behavior**
It should be possible to add an object as argument to the method enqueued as a simple background task.

**Additional context**
Other observations:
* `jobScheduler.enqueue(() -> myService.doWork("test1", 1));`  --> OK
* `jobScheduler.enqueue(() -> myService.doWork("test1", 1, "foo"));`  --> OK
* `String aStr="test1", Long id=1L; jobScheduler.enqueue(() -> myService.doWork(aStr, id));`  --> OK
* `String aStr="test1", Long id=1L; jobScheduler.enqueue(() -> myService.doWork(aStr, id,"foofoo"));`  --> OK
* `String aStr="test1", Long id=1L; String bStr="foofoo";  jobScheduler.enqueue(() -> myService.doWork(aStr,id,bStr));`  --> CRASH!!!!

Especially the last 2 sample use cases are very odd.

**Comment by rdehuyss** [comment]
Hmmm, that really baffles me as there are several unit tests covering this. I'll look into it asap.

**Comment by rdehuyss** [comment]
Hi @anowak-ct ,

Your first concern around the Pojo, I cannot reproduce... there are even a lot of tests around this: see the [JobDetailsAsmGeneratorTest](https://github.com/jobrunr/jobrunr/blob/master/core/src/test/java/org/jobrunr/jobs/details/JobDetailsAsmGeneratorTest.java#L255-L354).

Regarding the last 2 sample cases - that was indeed a bug and is fixed in commit 6e1065f4902861ecc53077fd6735e03260c2b9b9. I misinterpreted some of the JVM specification.

Could you setup a Github reproducing the issue with `MyPojo`?

**Comment by rdehuyss** [comment]
FYI - I even tested on the same JVM Vendor as you: See the [JdkTest](https://github.com/jobrunr/jobrunr/blob/master/tests/e2e-vm-jdk/src/test/java/org/jobrunr/tests/fromhost/JdkTest.java#?L65-L68) where I also use a simple pojo in the [TestService](https://github.com/jobrunr/jobrunr/blob/master/tests/e2e-vm-jdk/src/main/java/org/jobrunr/tests/e2e/services/TestService.java).

**Comment by anowak-ct** [comment]
Hi @rdehuyss 
Thanks for the quick response.  I will try to create a little sample project to reproduce the PoJo issue and let you know.

**Comment by rdehuyss** [comment]
Hi @anowak-ct - were you able to reproduce the issue?

**Comment by rdehuyss** [comment]
Resolved in [release v3.2.0](https://github.com/jobrunr/jobrunr/releases/tag/v3.2.0).

Please reopen the issue once you were able to setup a Github repo reproducing the issue.
